corehq/apps/app_manager/suite_xml/post_process/remote_requests.py

```diff
@@ -164,8 +166,10 @@ def get_post_relevant(self):
         if module_uses_smart_links(self.module):
             case_in_project = self._get_smart_link_rewind_xpath()
             return XPath.and_(case_not_claimed, case_in_project)
-        else:
+        if not case_search_sync_cases_on_form_entry_enabled_for_domain(self.domain):
             return case_not_claimed
+        else:
+            return None
```

Comment:
I think this logic isn't quite right since there is a return statement above. I think it should be something like this:
```python
case_not_claimed = X
case_in_project = Y
use_smart_links = Z
if not case_search_sync_cases_on_form_entry_enabled_for_domain(...):
    return case_in_project if use_smart_links else None
return XPath.and_(case_not_claimed, case_in_project) if use_smart_links else case_not_claimed


```

Reply:
thank you this resolved an issue I was looking into 🙌 https://github.com/dimagi/commcare-hq/pull/32554#:~:text=logic%20for%20get_post_relevant-,579c7f4,-Merge%20state